x-pack/plugins/task_manager/server/plugin.ts

```diff
@@ -3,92 +3,140 @@
  * or more contributor license agreements. Licensed under the Elastic License;
  * you may not use this file except in compliance with the Elastic License.
  */
-import { PluginInitializerContext, Plugin, CoreSetup, CoreStart } from 'src/core/server';
-import { Subject } from 'rxjs';
+import { PluginInitializerContext, Plugin, CoreSetup, Logger, CoreStart } from 'src/core/server';
 import { first } from 'rxjs/operators';
-import { TaskDictionary, TaskDefinition } from './task';
-import { TaskManager } from './task_manager';
+import { TaskDefinition } from './task';
+import { TaskPollingLifecycle } from './polling_lifecycle';
 import { TaskManagerConfig } from './config';
-import { Middleware } from './lib/middleware';
+import { createInitialMiddleware, addMiddlewareToChain, Middleware } from './lib/middleware';
 import { setupSavedObjects } from './saved_objects';
+import { TaskTypeDictionary } from './task_type_dictionary';
+import { FetchResult, SearchOpts, TaskStore } from './task_store';
+import { createManagedConfiguration } from './lib/create_managed_configuration';
+import { TaskScheduling } from './task_scheduling';
 
-export type TaskManagerSetupContract = Pick<
-  TaskManager,
-  'addMiddleware' | 'registerTaskDefinitions'
+export type TaskManagerSetupContract = { addMiddleware: (middleware: Middleware) => void } & Pick<
+  TaskTypeDictionary,
+  'registerTaskDefinitions'
 >;
 
 export type TaskManagerStartContract = Pick<
-  TaskManager,
-  'fetch' | 'get' | 'remove' | 'schedule' | 'runNow' | 'ensureScheduled'
->;
+  TaskScheduling,
+  'schedule' | 'runNow' | 'ensureScheduled'
+> &
+  Pick<TaskStore, 'fetch' | 'get' | 'remove'>;
 
 export class TaskManagerPlugin
   implements Plugin<TaskManagerSetupContract, TaskManagerStartContract> {
-  legacyTaskManager$: Subject<TaskManager> = new Subject<TaskManager>();
-  taskManager: Promise<TaskManager> = this.legacyTaskManager$.pipe(first()).toPromise();
-  currentConfig: TaskManagerConfig;
-  taskManagerId?: string;
-  config?: TaskManagerConfig;
+  private taskPollingLifecycle?: TaskPollingLifecycle;
+  private taskManagerId?: string;
+  private config?: TaskManagerConfig;
+  private logger: Logger;
+  private definitions: TaskTypeDictionary;
+  private middleware: Middleware = createInitialMiddleware();
 
   constructor(private readonly initContext: PluginInitializerContext) {
     this.initContext = initContext;
-    this.currentConfig = {} as TaskManagerConfig;
+    this.logger = initContext.logger.get('taskManager');
```

Comment:
The `taskManager` parameter can be removed. Platform has an item to do this for us otherwise: https://github.com/elastic/kibana/issues/79514.

```suggestion
    this.logger = initContext.logger.get();
```